### ISSUE TITLE
[v1.9.0] Release Tracker

### ISSUE THREAD

**Issue by driazati** [pytorch/pytorch#58518]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/1.9) for the 1.9.0 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 6/1): work on finalizing the release branch
* Phase 2 (after 6/1): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 6/1/21):**

Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent minor release (e.g. 1.8 for 1.9 release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3. Fixes to new features introduced in the most recent minor release (e.g. 1.8.1 for 1.9 release)
4. Test/CI fixes
5. Documentation improvements
6. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
7. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently @malfet, @seemethere, @gchanan, @driazati). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 6/1/21):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/1.9).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/1.9
    git checkout release/1.9
    git cherry-pick abcdef12345

    # Submit a PR based against 'release/1.9' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/1.9
    ```

    </details>
3. Make a request below with the following format:

```
Link to landed master PR (if applicable):
Link to release branch PR:
Criteria Category:
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

See #58519 for cherry pick status

*EDIT: Extending 1st phase deadline until Tue 6/1/21*

**Comment by zou3519** [comment]
Link to landed master PR (if applicable): n/a
Link to release branch PR: https://github.com/pytorch/pytorch/pull/58589
Criteria Category: Release branch specific changes (hiding feature that is not ready for beta/stable).

**Comment by jbschlosser** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/58403
Link to release branch PR: https://github.com/pytorch/pytorch/pull/58590
Criteria Category: Documentation

**Comment by Lilyjjo** [comment]
Link to landed master PR (if applicable): N/A 
Link to release branch PR: https://github.com/pytorch/pytorch/pull/58620 
Criteria Category: Release branch specific change (unhiding feature that is ready for beta but not for fb prod)

**Comment by cccclai** [comment]
Link to landed master PR (if applicable): #55734  #55733 #57888 
Link to release branch PR: #58630 (have three commits)
Criteria Category: Release branch specific change (unhiding feature that is ready for beta but not for fb prod)

**Comment by mcarilli** [comment]
master PR: https://github.com/pytorch/pytorch/pull/58679 (approved and should be merged soon, filing this now so I don't forget)
release/1.9 PR: https://github.com/pytorch/pytorch/pull/58683
Criteria: Fixes a crash

**Comment by ngimel** [comment]
Master PR: #58468
Release/1.9 PR #58714
Criteria: fixes a regression from 1.8

**Comment by bdhirsh** [comment]
Master PR: #57510, #57361, #55796, #56597, #55795 (split into small pieces but landed together)
Release/1.9 PR: #58717

Criteria: These are corresponding changes that are required for any XLA side change, here: https://github.com/pytorch/xla/pull/2915. They didn't make it into the release cut, but the XLA side changes have been in master and will end up in the XLA-side release cut.

An alternative would be to revert https://github.com/pytorch/xla/commit/3e3704f23144793e88e31574c7413ffb310d9c7d from the XLA release branch once it's generated.

**Comment by nSircombe** [comment]
Master PR: #55913
Release/1.9 PR: #58746
Criteria: improved support for oneDNN on AArch64 when building from src.

**Comment by malfet** [comment]
Master PR: https://github.com/pytorch/pytorch/pull/58763
Release PR: https://github.com/pytorch/pytorch/pull/58763
Criteria: Fixes Python-3.9 CI

**Comment by walterddr** [comment]
Master PR: #58751
Release PR: #58752
Criteria: Fixes unrelated, release branch CI failures

**Comment by ejguan** [comment]
Master PR: #58450
Release PR: #58848
Criteria: Fixes to new features introduced in the most recent minor release (Fix bug for the new feature (typing system for DataPipe) introduced in 1.9)

**Comment by soulitzer** [comment]
Link to landed master PR (if applicable): #58513
Link to release branch PR: #58939
Criteria Category: Documentation

**Comment by jbschlosser** [comment]
Master PR: #58648
Release PR: #58940
Criteria: Low-risk, popular new function / module (see [here](https://twitter.com/DigantaMisra1/status/1397259602622771200))

**Comment by jbschlosser** [comment]
Master PR: #58809
Release PR: #58958
Criteria: Documentation

**Comment by ezyang** [comment]
Release PR: https://github.com/pytorch/pytorch/pull/58994
Criteria Category: Critical fixes (new feature does not work, this PR disables it)

**Comment by malfet** [comment]
Release PR: https://github.com/pytorch/pytorch/pull/59008
Master PR: None
Criteria Category: Fixes compilation regressions caused by #58940

**Comment by jbschlosser** [comment]
Master PR: #59021
Release PR: #59044
Criteria Category: Documentation

**Comment by jbschlosser** [comment]
Master PR: #59025
Release PR: #59045
Criteria Category: Documentation

**Comment by ngimel** [comment]
Master pr: #58800
release pr: #59048
Criteria Category: Regression fix

**Comment by ngimel** [comment]
Master pr: #59003
release pr: #59055 
Criteria Category: Regression fix

**Comment by bhosmer** [comment]
Master PR: #58727
Release PR: #59072
Criteria Category: Regression fix
cc @jbschlosser

**Comment by soulitzer** [comment]
Master pr: #59047
Release pr: #59085
Criteria Category: Documentation

**Comment by driazati** [comment]
Master PR: n/a
Release PR: #59093
Criteria Category: Regression fix

**Comment by anjali411** [comment]
Master PR: https://github.com/pytorch/pytorch/pull/59027
Release PR: https://github.com/pytorch/pytorch/pull/59105
Criteria: Add underscore to internal API names

**Comment by malfet** [comment]
Master PR: #57400 
Release PR: #59116 
Criteria: ROCm specific fixes

**Comment by malfet** [comment]
Master PR: #58996 
Release PR: #59118
Criteria: Low-risk fix for null-pointer dereference

**Comment by malfet** [comment]
Master PR: #58437
Release PR: #59119
Criteria: Reduce binary size of CUDA-11 builds

**Comment by malfet** [comment]
Master PR: #58459
Release PR: #59120
Criteria: Reduce binary size of CUDA-11 builds

**Comment by malfet** [comment]
Master PR: #58458
Release PR: #59121
Criteria: collect_env tool improvements

**Comment by ngimel** [comment]
@driazati #59055 is not merged

**Comment by pbelevich** [comment]
Please build PyTorch with `USE_GLOO_WITH_OPENSSL=1` to support TLS transport in Gloo

**Comment by rohan-varma** [comment]
Master PR: https://github.com/pytorch/pytorch/pull/58702
Release PR: https://github.com/pytorch/pytorch/pull/59266
Criteria: Critical fixes

**Comment by SplitInfinity** [comment]
Master PR: #58573 
Release PR: #59272 
Criteria: Fixes to new features introduced in the most recent minor release (`torch.package`)

**Comment by SplitInfinity** [comment]
Master PR: #58572
Release PR: #59273
Criteria: Fixes to new features introduced in the most recent minor release (`torch.package`)

**Comment by driazati** [comment]
Master PR: n/a
Release PR: #59274
Criteria: release branch specific changes

**Comment by driazati** [comment]
Master PR: #59236
Release PR: #59275
Criteria: Feature fixes

**Comment by malfet** [comment]
Master PR: #59158 and #58701
Release PR: https://github.com/pytorch/pytorch/pull/59339
Criteria: CI fixes (disable failing tests)

**Comment by IvanKobzarev** [comment]
Master PR: https://github.com/pytorch/pytorch/pull/59276
Release PR: https://github.com/pytorch/pytorch/pull/59341
Criteria: Fixes to new features introduced in the most recent minor release (lite interpreter)

**Comment by Minipeps** [comment]
Master PR: #45736
Release PR: #59367
Criteria: Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries

**Comment by zhouzhuojie** [comment]
Master PR:  #59352 
Release PR: #59452
Criteria: Bug fixes and 1.9 milestone issue (https://github.com/pytorch/pytorch/issues/58545)

**Comment by seemethere** [comment]
Master PR: https://github.com/pytorch/pytorch/pull/59413
Release PR: https://github.com/pytorch/pytorch/pull/59462
Criteria: Fixes crash for CUDA 10.2 binaries when throwing exceptions

**Comment by malfet** [comment]
Master PR:  #59428
Release PR: https://github.com/pytorch/pytorch/pull/59463
Criteria: AARCH64 build fixes

**Comment by malfet** [comment]
Master PR: #59361
Release PR: https://github.com/pytorch/pytorch/pull/59470
Criteria: AARCH64 accuracy fixes

**Comment by malfet** [comment]
Master PR: #59490
Release PR: https://github.com/pytorch/pytorch/pull/59500
Criteria: Regression from previous release (Fixes crash for CUDA-11 binaries similar to #59462)

**Comment by malfet** [comment]
Master PR: #59467
Release PR: https://github.com/pytorch/pytorch/pull/59501
Criteria: Error message update

**Comment by malfet** [comment]
Master PR: #59484
Release PR: #59502
Criteria: CI fixes

**Comment by ezyang** [comment]
Master PR: https://github.com/pytorch/pytorch/pull/59633
Release PR: pending
Criteria: Fixes to new features introduced in the most recent minor release (namely `inference_mode`)

**Comment by cccclai** [comment]
> Master PR: #59642
> Release PR: #59672
> Criteria: Critical fix for improper mobile and torch.package serialization
> 
> Are new PR still being added to the cut?

Sorry for the late notice. The issue was discovered last Thursday. The pr fixes the flakiness when load/save models multiple times for two new features in 1.9 (lite interpreter and torch.package). It will nice to have it in.

**Comment by malfet** [comment]
Master PR: #58287, #59721, #59744 and #59802
Release PR: https://github.com/pytorch/pytorch/pull/59873
Criteria: Fixes cuDNN performance on A100

**Comment by SplitInfinity** [comment]
Master PR: Various
Release PR: #59866 
Criteria: Documentation for new feature in this release

**Comment by SplitInfinity** [comment]
Master PR: #59959
Release PR: #59976 
Criteria: Documentation for new feature in this release

**Comment by mberr** [comment]
This may be a very minor issue, but some labels linked in the issue template seem to be outdated, e.g. for `topic: regression` from

> Fixes to regressions against the most recent minor release (e.g. 1.8 for 1.9 release; see [topic: regression issue list](https://github.com/pytorch/pytorch/issues?q=is:issue%20is:open%20label:%22topic:%20regression%22/))

there is no issues labeled with it. However, there is a label of similar name, `module: regression` which has [some issues](https://github.com/pytorch/pytorch/labels/module%3A%20regression).

**Comment by seemethere** [comment]
Thanks everyone for all of the help regarding the 1.9.0 release!

Going to go ahead and close this issue since the 1.9.0 release has already happened

**Comment by suo** [comment]
Docs only change, hopefully we can regenerate the docs:
Master PR: #60113
Cherry pick PR: https://github.com/pytorch/pytorch/pull/60117

Criteria: Documentation for new feature in this release
